bitcoin/src/psbt/mod.rs

```diff
@@ -2273,12 +2375,73 @@ mod tests {
 
     #[test]
     #[cfg(feature = "rand-std")]
-    fn sign_psbt() {
-        use crate::address::script_pubkey::ScriptBufExt as _;
-        use crate::bip32::{DerivationPath, Fingerprint};
-        use crate::witness_version::WitnessVersion;
-        use crate::WitnessProgram;
+    fn hashmap_can_sign_taproot() {
+        let (priv_key, pk, secp) = gen_keys();
+        let internal_key: XOnlyPublicKey = pk.inner.into();
+
+        let tx = Transaction {
+            version: transaction::Version::TWO,
+            lock_time: locktime::absolute::LockTime::ZERO,
+            input: vec![TxIn::EMPTY_COINBASE],
+            output: vec![TxOut { value: Amount::ZERO, script_pubkey: ScriptBuf::new() }],
+        };
 
+        let mut psbt = Psbt::from_unsigned_tx(tx).unwrap();
+        psbt.inputs[0].tap_internal_key = Some(internal_key);
+        psbt.inputs[0].witness_utxo = Some(transaction::TxOut {
+            value: Amount::from_sat_unchecked(10),
```

Comment:
In 3d87885eafee41e7beb9e3c8e6bb04ae8a39bff5:

Sorry, you need to rebase because of this (the merge commit does not pass my local CI). `from_sat_unchecked` was removed. You can use `from_sat_u32` instead.